### PR TITLE
fix(gatsby-plugin-offline): use networkFirst caching for page-data.json files

### PR DESCRIPTION
**Pull request by Moocar**
## Description

`page-data.json` files are not content hashed. Their value can change between builds and the filename will remain the same. Therefore, in the offline plugin, we should use [networkFirst](https://developers.google.com/web/fundamentals/instant-and-offline/offline-cookbook/#network-falling-back-to-cache) strategy.

## Related Issues

- https://github.com/gatsbyjs/gatsby/pull/13004

### CODE REVIEW

File: packages/gatsby-plugin-offline/src/gatsby-node.js [hunk 1]
```diff
@@ -90,6 +90,11 @@ exports.onPostBuild = (args, pluginOptions) => {
         urlPattern: /(\.js$|\.css$|static\/)/,
         handler: `cacheFirst`,
       },
+      {
+        // page-data.json files are not content hashed
+        urlPattern: /^https:?:*page-data\.json/,
```
Comment (KyleAMathews) [inline packages/gatsby-plugin-offline/src/gatsby-node.js]: This didn't seem to work when I tested it — something like this did https://regex101.com/r/WgnMvi/1

Comment (Moocar) [reply packages/gatsby-plugin-offline/src/gatsby-node.js]: Ahh interesting, I thought I saw it working locally, but I'll try again. With these changes.

Comment (Moocar) [reply packages/gatsby-plugin-offline/src/gatsby-node.js]: Ok, fixed and actually tested changing `page-data.json` to make sure the new results are shown on the first network request

**Comment by Moocar** [comment]
Thanks for testing! I don't mind about the `$` suffix. The regex is already quite specific. Would it help with performance at all?

**Comment by KyleAMathews** [comment]
Yeah, seems fine without it. Merging!

**Pull request merged**